app/controllers/orders_controller.rb

```diff
@@ -1,6 +1,11 @@
 class OrdersController < ApplicationController
   def index
-    @orders = current_user.orders
+    if current_user
+      @orders = current_user.orders
+    else
+      flash[:non_user] = "You must be logged in to view your orders"
+      render file: 'public/404'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.